trino/sqlalchemy/dialect.py

```diff
@@ -0,0 +1,301 @@
+            raise
+
+    def has_schema(self, connection: Connection, schema: str) -> bool:
+        query = f"SHOW SCHEMAS LIKE '{schema}'"
+        try:
+            res = connection.execute(sql.text(query))
+            return res.first() is not None
+        except error.TrinoQueryError as e:
+            if e.error_name in (
+                    error.TABLE_NOT_FOUND,
+                    error.SCHEMA_NOT_FOUND,
+                    error.CATALOG_NOT_FOUND,
+            ):
+                return False
+            raise
+
+    def has_table(self, connection: Connection,
+                  table_name: str, schema: str = None) -> bool:
+        query = 'SHOW TABLES'
+        if schema:
+            query = f'{query} FROM {self.identifier_preparer.quote_identifier(schema)}'
+        query = f"{query} LIKE '{table_name}'"
+        try:
+            res = connection.execute(sql.text(query))
+            return res.first() is not None
+        except error.TrinoQueryError as e:
+            if e.error_name in (
+                    error.TABLE_NOT_FOUND,
+                    error.SCHEMA_NOT_FOUND,
+                    error.CATALOG_NOT_FOUND,
+                    error.MISSING_SCHEMA_NAME,
+            ):
+                return False
+            raise
+
+    def has_sequence(self, connection: Connection,
+                     sequence_name: str, schema: str = None) -> bool:
+        """Trino has no support for sequence. Returns False indicate that given sequence does not exists."""
+        return False
+
+    def _get_server_version_info(self, connection: Connection) -> Tuple[int, ...]:
+        query = 'SELECT version()'
+        res = connection.execute(sql.text(query))
+        version = res.scalar()
+        return tuple([version])
+
+    def _get_default_schema_name(self, connection: Connection) -> Optional[str]:
+        dbapi_connection: trino_dbapi.Connection = connection.connection
+        return dbapi_connection.schema
+
+    def do_rollback(self, dbapi_connection):
+        if dbapi_connection.transaction is not None:
+            dbapi_connection.rollback()
+
+    def do_begin_twophase(self, connection: Connection, xid):
+        pass
+
+    def do_prepare_twophase(self, connection: Connection, xid):
+        pass
+
+    def do_rollback_twophase(self, connection: Connection, xid,
+                             is_prepared: bool = True, recover: bool = False) -> None:
+        pass
+
+    def do_commit_twophase(self, connection: Connection, xid,
+                           is_prepared: bool = True, recover: bool = False) -> None:
+        pass
+
+    def do_recover_twophase(self, connection: Connection) -> None:
+        pass
+
+    def set_isolation_level(self, dbapi_conn: trino_dbapi.Connection, level) -> None:
+        dbapi_conn._isolation_level = getattr(trino_dbapi.IsolationLevel, level)
+
+    def get_isolation_level(self, dbapi_conn: trino_dbapi.Connection) -> str:
+        level_names = ['AUTOCOMMIT',
```

Comment:
Should the dbapi declare the levels as an exported field instead of repeating them here?

Reply:
Yeah, I just refactor `IsolationLevel` to `Enum`. So no need to define `level_names` here :+1:

Reply:
Reminder - looks like we still have `level_names` used here.

Reply:
Seems this is fixed in a later commit.